inksdk/inksample/src/main/java/com/microsoft/device/inksample/MainActivity.kt

```diff
@@ -44,25 +46,47 @@ class MainActivity : AppCompatActivity() {
     }
 
     fun setRed(@Suppress("UNUSED_PARAMETER")view: View) {
+        resetPaintHandler()
         inkView.color = Color.RED
     }
 
     fun setGreen(@Suppress("UNUSED_PARAMETER")view: View) {
+        resetPaintHandler()
         inkView.color = Color.GREEN
     }
 
     fun setBlue(@Suppress("UNUSED_PARAMETER")view: View) {
+        resetPaintHandler()
         inkView.color = Color.BLUE
     }
 
     fun setBlack(@Suppress("UNUSED_PARAMETER")view: View) {
+        resetPaintHandler()
         inkView.color = Color.BLACK
     }
 
     fun setYellow(@Suppress("UNUSED_PARAMETER")view: View) {
+        resetPaintHandler()
         inkView.color = Color.YELLOW
     }
 
+    fun setRainbow(@Suppress("UNUSED_PARAMETER")view: View) {
+        inkView.dynamicPaintHandler = RainbowPaintHandler()
+        fancySwitch.isChecked = false
+    }
+    /**
+     * Required because some colors are set on the InkView, but
+     * rainbow ink is controlled via a paint handler (which is also
+     * how the pressure handler works, but it modifies color)
+     */
+    fun resetPaintHandler() {
```

Comment:
Even when `inkView` and `fancySwitch` are global variables that can be accessed along the class whenever we want, we maybe want to address the new reset-value inkView will get with a returned value so we don't hide the functionality of what happens when we call `resetPaintHandler`. For example something like:

```
fun setYellow(@Suppress("UNUSED_PARAMETER")view: View) {
        inkView.dynamicPaintHanlder = resetPaintHandler(fancySwitch.isChecked)
        inkView.color = Color.YELLOW
    }

fun resetPaintHandler (isFancySwitchChecked : Boolean) : FancyPaintHandler? =
       return if (isFancySwitchChecked) {
            FancyPaintHandler()
        } else {
            null
        }
}
```

Reply:
@CesarValiente I like this approach - I propose we implement in another PR so I can ship this today 😉
Thanks for the feedback. The fix will be coming soon!

Reply:
Sure!! not a blocker at all! ;-)